### ISSUE TITLE
which version of omnetpp and INET I have to install for using your latest Denacast code

### ISSUE THREAD

**Issue by krasper** [aarizaq/Denacast#2]

**Comment by aarizaq** [comment]
this version uses inet 2.x and omnet 4.6

**Comment by krasper** [comment]
when i run denacast , CDN-P2P config, it asks for parameter 'InetUnderlayNetwork2.xxxType' and the default are "". After OK to all parameters , It occurs an error. 

 Error in module (cModule) InetUnderlayNetwork2.overlayAccessRouter[0](id=5) during network setup: Submodule tier2: no module type named `' found that implements module interface oversim.common.ITier (not in the loaded NED files?)

What parameters i have to input? Moreover , there are some error in denacast.ini file like the attached picture.
How to solve these problems, thank you
![screen shot 2559-11-20 at 17 26 46](https://cloud.githubusercontent.com/assets/6574257/20462127/8aea02dc-af46-11e6-9458-d5547a6cd096.png)

**Comment by aarizaq** [comment]
But , can you run the simulation?
The errors show by the ide can not be real errors

De: krasper [mailto:notifications@github.com]
Enviado el: domingo, 20 de noviembre de 2016 11:02
Para: aarizaq/Denacast Denacast@noreply.github.com
CC: Alfonso aarizaq_m@hotmail.com; Comment comment@noreply.github.com
Asunto: Re: [aarizaq/Denacast] which version of omnetpp and INET I have to install for using your latest Denacast code (#2)

when i run denacast , CDN-P2P config, it asks for parameter 'InetUnderlayNetwork2.xxxType' and the default are "". After OK to all parameters , It occurs an error.

What parameters i have to input? Moreover , there are some error in denacast.ini file like the attached picture.
How to solve these problems, thank you
[screen shot 2559-11-20 at 16 59 15]https://cloud.githubusercontent.com/assets/6574257/20462038/f47296be-af42-11e6-824b-ae6b18075215.png

**Comment by krasper** [comment]
no, it cannot run. if don't use the IDE how can I run Denacast. pls give me an install and usage instruction.

**Comment by aarizaq** [comment]
You should use oversim-20121206, it includes some fixed for omnet 4.6

De: krasper [mailto:notifications@github.com]
Enviado el: martes, 22 de noviembre de 2016 14:02
Para: aarizaq/Denacast Denacast@noreply.github.com
CC: Alfonso aarizaq_m@hotmail.com; Comment comment@noreply.github.com
Asunto: Re: [aarizaq/Denacast] which version of omnetpp and INET I have to install for using your latest Denacast code (#2)

no, it cannot run. if don't use the IDE how can I run Denacast. pls give me an install and usage instruction.

**Comment by krasper** [comment]
Therefore , I have to install 3 libraries in my project , inetmanet-2.0,denacast, and oversim-20121206 ?

**Comment by aarizaq** [comment]
No, you only need inetmanet-2.0 and oversim-20121206

De: krasper [mailto:notifications@github.com]
Enviado el: martes, 22 de noviembre de 2016 14:22
Para: aarizaq/Denacast Denacast@noreply.github.com
CC: Alfonso aarizaq_m@hotmail.com; Comment comment@noreply.github.com
Asunto: Re: [aarizaq/Denacast] which version of omnetpp and INET I have to install for using your latest Denacast code (#2)

Therefore , I have to install 3 libraries in my project , inetmanet-2.0,denacast, and oversim-20121206 ?

**Comment by krasper** [comment]
But I want to use CDN-P2P feature in Denacast Project , How can i use CDN-P2P in oversim-20121206 ?

**Comment by aarizaq** [comment]
Yes, you can, I have included the specific denacast code in oversim

De: krasper [mailto:notifications@github.com]
Enviado el: martes, 22 de noviembre de 2016 18:19
Para: aarizaq/Denacast Denacast@noreply.github.com
CC: Alfonso aarizaq_m@hotmail.com; Comment comment@noreply.github.com
Asunto: Re: [aarizaq/Denacast] which version of omnetpp and INET I have to install for using your latest Denacast code (#2)

But I want to use CDN-P2P feature in Denacast Project , How can i use CDN-P2P in oversim-20121206 ?

**Comment by shivanshu123** [comment]
In ubuntu 18.04 ,Do i need to install omnet-4.2.2 version to run this project or the latest version can help,as the old version is giving some unresoluable errors.
Because during compilation,I found an error 
 sim_std_m.h:5:0: error: unterminated #ifndef
 #ifndef _SIM_STD_M_H_

but at the end of file ,#endif is already present.

**Comment by aarizaq** [comment]
Denacast is a quite old, the lates version of oversim is better and include also the video part

Enviado desde Correo<https://go.microsoft.com/fwlink/?LinkId=550986> para Windows 10

________________________________
De: shivanshu123 <notifications@github.com>
Enviado: Thursday, April 16, 2020 1:32:44 PM
Para: aarizaq/Denacast <Denacast@noreply.github.com>
Cc: Alfonso <aarizaq_m@hotmail.com>; Comment <comment@noreply.github.com>
Asunto: Re: [aarizaq/Denacast] which version of omnetpp and INET I have to install for using your latest Denacast code (#2)


In ubuntu 18.04 ,Do i need to install omnet-4.2.2 version to run this project or the latest version can help,as the old version is giving some unresoluable errors.

**Comment by shivanshu123** [comment]
Can someone please provide the output graphs for it as i'm currently unable to run it and want to analysis the output graphs of frame lost or burst and distortion etc.

**Comment by shivanshu123** [comment]
> when i run denacast , CDN-P2P config, it asks for parameter 'InetUnderlayNetwork2.xxxType' and the default are "". After OK to all parameters , It occurs an error.
> 
> Error in module (cModule) InetUnderlayNetwork2.overlayAccessRouter[0](id=5) during network setup: Submodule tier2: no module type named `' found that implements module interface oversim.common.ITier (not in the loaded NED files?)



I' m getting same error, did u solve the problem , if yes then pls provide a solution what should I do?

**Comment by aarizaq** [comment]
I don't remember, but is it not better to use Oversim? the code is more completed that Denacast and there is a version that is been maintained and adaptated to a more moderm version of inet 
https://github.com/inet-framework/oversim

**Comment by shivanshu123** [comment]
@aarizaq ,Can you please suggest how to  get graph in omnetpp as i'm unable to produce graph using your project ,Do i need something else as i'm getting the simulation correct but unable to produce graph .Please share your graph with us.Please Help.

**Comment by aarizaq** [comment]
It is possible that the code could have incompatibilities, this code is not maintained and the principal code has ben modified.
